ui/dashboards/src/components/Variables/VariableEditor.tsx

```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
-                <TableBody>
-                  {variableDefinitions.map((v, idx) => (
-                    <TableRow key={v.spec.name}>
-                      <TableCell component="th" scope="row">
-                        <Switch
-                          checked={v.spec.display?.hidden !== true}
-                          onChange={(e) => {
-                            toggleVariableVisibility(idx, e.target.checked);
-                          }}
-                        />
-                      </TableCell>
-                      <TableCell component="th" scope="row" sx={{ fontWeight: 'bold' }}>
-                        {v.spec.name}
-                      </TableCell>
-                      <TableCell>{v.kind}</TableCell>
-                      <TableCell align="right">
-                        <IconButton onClick={() => changeVariableOrder(idx, 'up')} disabled={idx === 0}>
-                          <ArrowUp />
-                        </IconButton>
-                        <IconButton
-                          onClick={() => changeVariableOrder(idx, 'down')}
-                          disabled={idx === variableDefinitions.length - 1}
-                        >
-                          <ArrowDown />
-                        </IconButton>
-
-                        <IconButton onClick={() => setVariableEditIdx(idx)}>
-                          <PencilIcon />
-                        </IconButton>
-                        <IconButton onClick={() => removeVariable(idx)}>
-                          <TrashIcon />
-                        </IconButton>
-                      </TableCell>
-                    </TableRow>
-                  ))}
-                </TableBody>
-              </Table>
-            </TableContainer>
-            <Box display="flex">
-              <Button onClick={addVariable} variant="contained">
-                Add New Variable
+          <Box
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              padding: (theme) => theme.spacing(1, 2),
+              borderBottom: (theme) => `1px solid ${theme.palette.grey[100]}`,
+            }}
+          >
+            <Typography variant="h2">Template Variables</Typography>
+            <Stack direction="row" spacing={1} sx={{ marginLeft: 'auto' }}>
+              <Button
+                disabled={props.variableDefinitions === variableDefinitions || !validation.isValid}
+                variant="contained"
+                onClick={() => {
+                  props.onChange(variableDefinitions);
+                }}
+              >
+                Apply
+              </Button>
+              <Button
+                variant="outlined"
+                onClick={() => {
+                  props.onCancel();
+                }}
+              >
+                Cancel
               </Button>
-            </Box>
-          </Stack>
+            </Stack>
+          </Box>
+          <Box padding={2}>
```

Comment:
Added this `Box` for correct padding, most of the remaining changes in this file are indentation unfortunately